crates/pera-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -189,39 +219,67 @@ impl DWalletMPCManager {
 
     fn handle_event(&mut self, event: Event, session_info: SessionInfo) -> DwalletMPCResult<()> {
         self.outputs_verifier.handle_new_event(&session_info);
-        if let Ok((party, auxiliary_input, session_info)) = from_event(
-            &event,
-            &self,
-            authority_name_to_party_id(&self.epoch_store()?.name, &*self.epoch_store()?)?,
-        ) {
-            self.push_new_mpc_session(auxiliary_input, party, session_info)?;
-        };
+        let (public_input, private_input) = session_input_from_event(&event, &self)?;
+        self.push_new_mpc_session(public_input, private_input, session_info)?;
         Ok(())
     }
 
-    // todo(zeev): doc this.
-    pub fn get_decryption_share(&self) -> DwalletMPCResult<DecryptionKeyShare> {
+    pub(super) fn get_protocol_public_parameters(
+        &self,
+        key_scheme: DWalletMPCNetworkKeyScheme,
+        key_version: u8,
+    ) -> DwalletMPCResult<Vec<u8>> {
+        if let Some(self_decryption_share) = self.epoch_store()?.dwallet_mpc_network_keys.get() {
+            return self_decryption_share.get_protocol_public_parameters(key_scheme, key_version);
+        }
+        Err(DwalletMPCError::TwoPCMPCError(
+            "Decryption share not found".to_string(),
+        ))
+    }
+
+    pub(super) fn get_decryption_public_parameters(
```

Comment:
```suggestion
    pub(super) fn get_decryption_key_share_public_parameters(
```
You might think it helps using these shortened names, but it has the opposite effect of confusing auditors. Please use my original names every where